src/trio/_tests/type_tests/raisesgroup.py

```diff
@@ -92,6 +93,19 @@ def check_matcher_transparent() -> None:
     assert_type(e.value, BaseExceptionGroup[ValueError])
 
 
+def check_matcher_tostring() -> None:
+    assert str(Matcher(ValueError)) == "Matcher(ValueError)"
+    assert str(Matcher(match="[a-z]")) == "Matcher(match='[a-z]')"
+    pattern_no_flags = re.compile("noflag", 0)
+    assert str(Matcher(match=pattern_no_flags)) == "Matcher(match='noflag')"
+    pattern_flags = re.compile("noflag", re.IGNORECASE)
+    assert str(Matcher(match=pattern_flags)) == f"Matcher(match={pattern_flags!r})"
+    assert (
+        str(Matcher(ValueError, match="re", check=bool))
+        == f"Matcher(ValueError, match='re', check={bool!r})"
```

Comment:
Is it intended that check is set to boolean constructor? I think you intended to set check to either True or False as with a pytest.parametrize. If so, don't use `bool` as a variable name, because it would be shadowing a python builtin.

Reply:
Yes, I was using bool as an arbitrary function. Maybe should have used something else.

Reply:
I think it's fine

Reply:
...though why is this in `type_tests`

Reply:
... whoops. That explains the coverage.